Android/ExposureNotification/source/PlayServicesNearby/Additions/ExposureNotificationClient.cs

```diff
@@ -1,45 +1,42 @@
-﻿using System;
+﻿using Android.Runtime;
+using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Threading.Tasks;
-using Android.Content;
-using Android.Hardware;
-using Android.Runtime;
 using AndroidTask = Android.Gms.Tasks.Task;
 
 namespace Android.Gms.Nearby.ExposureNotification
 {
 	public partial interface IExposureNotificationClient
 	{
-		public Task Start(ExposureConfiguration config)
+		public Task StartAsync(ExposureConfiguration config)
 			=> NativeStart(config).CastTask();
 
-		public Task Stop()
+		public Task StopAsync()
 			=> NativeStop().CastTask();
 
-		public Task<ExposureInformation> GetExposureInformation()
-			=> NativeExposureInformation().CastTask<ExposureInformation>();
+		public async Task<IList<ExposureInformation>> GetExposureInformationAsync()
+			=> await NativeExposureInformation().CastTask<JavaList<ExposureInformation>>();
 
-		public Task<ExposureSummary> GetExposureSummary()
+		public Task<ExposureSummary> GetExposureSummaryAsync()
 			=> NativeExposureSummary().CastTask<ExposureSummary>();
 
-		public Task<Java.Lang.Integer> GetMaxDiagnosisKeyCount()
-			=> NativeGetMaxDiagnosisKeyCount().CastTask<Java.Lang.Integer>();
+		public async Task<int> GetMaxDiagnosisKeyCountAsync()
+			=> (int)await NativeGetMaxDiagnosisKeyCount().CastTask<Java.Lang.Integer>();
 
-		public Task<Java.Lang.Boolean> IsEnabled()
-			=> NativeIsEnabled().CastTask<Java.Lang.Boolean>();
+		public async Task<bool> IsEnabledAsync()
+			=> (bool)await NativeIsEnabled().CastTask<Java.Lang.Boolean>();
 
-		public Task ProvideDiagnosisKeys(List<TemporaryExposureKey> keys)
+		public Task ProvideDiagnosisKeysAsync(IList<TemporaryExposureKey> keys)
 			=> NativeProvideDiagnosisKeys(keys).CastTask();
 
-		public Task ResetAllData()
+		public Task ResetAllDataAsync()
 			=> NativeResetAllData().CastTask();
 
-		public Task ResetTemporaryExposureKey()
+		public Task ResetTemporaryExposureKeyAsync()
 			=> NativeResetTemporaryExposureKey().CastTask();
 
-		public Task<JavaList<TemporaryExposureKey>> GetTemporaryExposureKeyHistory()
-			=> NativeStop().CastTask<JavaList<TemporaryExposureKey>>();
+		public async Task<IList<TemporaryExposureKey>> GetTemporaryExposureKeyHistoryAsync()
```

Comment:
Removed the `JavaList<T>`. Nobody likes that. Anyway, interfaces use interfaces. I think so at least... :)

Reply:
Not sure, will the cast work properly for this? I kind of err'd on the side of not and getting a version out.

Reply:
I cast first to JavaList. This implements IList, so this is not actually a cast.